### PR TITLE
add pkg-config file

### PR DESCRIPTION
**Pull request by jprjr**
Hi there - PR builds and installs pkg-config file for libmgba

### CODE REVIEW

File: CMakeLists.txt
Comment (endrift) [inline CMakeLists.txt:1011]:
```suggestion
	NAME "${PROJECT_NAME}"
	DESC "${PROJECT_NAME} library"
```

File: src/platform/cmake/MgbaPkgConfig.cmake
Comment (endrift) [inline src/platform/cmake/MgbaPkgConfig.cmake:36]: If these are hardcoded, you should just stick it directly in the .pc.in instead.

**Review by endrift — CHANGES_REQUESTED** [review]
The project is called mGBA, not MGBA or Mgba, and this is liable to change when medusa gets merged in. It looks like you just took a stock pkg-config cmake generator and willy-nilly stuffed it in, while you could really just use `configure_file` directly if you took a little bit of extra care with what you were doing. Most of the lines in that CMake file are comments or useless (the string replace bits can be completely removed if you set up PKGCFG_DEFINES properly when passing it in).

Adding a pkgconf config is a good idea, but this PR has a bit to go before it can be merged.

File: src/platform/cmake/MgbaPkgConfig.cmake
Comment (jprjr) [reply src/platform/cmake/MgbaPkgConfig.cmake]: Corrected in 8f9617d6948132830ee0dd5efc98b76ee7d7077e

**Comment by endrift** [comment]
This looks fine now, though I'll need to review it a bit more closely later.

**Comment by jprjr** [comment]
One thing I'm unsure about is what cflags/defines should be added to the pkg-config file (for a downstream user), vs which ones will strictly be internal flags. Right now it's not adding any to the file.

**Comment by endrift** [comment]
Most of those should already be in the configured flags.h file.

File: CMakeLists.txt
Comment (endrift) [inline CMakeLists.txt:226]: This should probably be done by looping over `OS_LIB` at the end of the platform support block (right before the feature dependencies block)

File: CMakeLists.txt
Comment (endrift) [inline CMakeLists.txt:352]: Not 100% sure, but it's probably supposed to be this:
```suggestion
			set(PKGCFG_LDFLAGS_PRIV "${PKGCFG_LDFLAGS_PRIV} -pthread")
```

**Review by endrift — COMMENTED** [review]
This is _probably_ fine, but two tweak suggestions. The former is most important on Windows (where very few people would care about pkg-config anyway), and the latter may or may not be necessary (I'm not sure).

File: CMakeLists.txt
Comment (jprjr) [reply CMakeLists.txt:352]: pkg-config `Requires.private` field is only used when linking against a static library (or more accurate, when you use the `--static` flag in your call to `pkg-config`). If pthreads isn't being used directly by the consuming app/library, then `-lpthread` should be fine to resolve missing symbols.

With dynamic libraries it's a non-issue, since the dynamic lib will already be linked against `libpthread`

Comment (endrift) [reply CMakeLists.txt:371]: I'm aware of the role of `Libs.private`, I was just worried that `-lpthread` wouldn't set all of the correct linker flags, since the objects being pulled in from the static lib haven't been linked prior.

File: CMakeLists.txt
Comment (endrift) [inline CMakeLists.txt:1031]: Typo I didn't notice earlier:
```suggestion
configure_file(${CMAKE_CURRENT_SOURCE_DIR}/res/mgba.pc.in ${CMAKE_CURRENT_BINARY_DIR}/mgba.pc @ONLY)
```

**Review by endrift — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (endrift) [reply CMakeLists.txt:371]: Digging through my pkgconfig directory as-is, it seems that the libraries installed are pretty evenly divided on this one, so I'm not gonna press it.

File: CMakeLists.txt
Comment (endrift) [reply CMakeLists.txt:245]: Are you planning to do anything about this comment?

**Comment by jprjr** [comment]
Yes, though I'm not sure when I'll be able to due to my schedule (hopefully this weekend I'll have some time). I plan to loop over the OS_LIB variable, like suggested.

On March 19, 2021 2:20:10 PM EDT, Vicki Pfau ***@***.***> wrote:
***@***.*** commented on this pull request.
>
>
>
>> @@ -220,6 +223,7 @@ endif()
> 
> if(NOT HAIKU AND NOT MSVC AND NOT PSP2)
> 	set(M_LIBRARY m)
>+	set(PKGCFG_LDFLAGS_PRIV "${PKGCFG_LDFLAGS_PRIV} -lm")
>
>Are you planning to do anything about this comment?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/mgba-emu/mgba/pull/2064#discussion_r597889096

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jprjr** [comment]
Hey I'll just be really up-front. I'm not enjoying this. You're free to use what I've written, but I won't be contributing/interacting again. I'm sorry for wasting anybody's time.

**Pull request closed without merging**
